addons/web_editor/static/lib/odoo-editor/src/utils/sanitize.js

```diff
@@ -187,8 +187,13 @@ class Sanitize {
 
             // Ensure elements which should not contain any content are tagged
             // contenteditable=false to avoid any hiccup.
+            // TODO: Some i.fa must have the contenteditable attribute set to
+            // true because they are descendants of an o_not_editable but the
+            // user should still be able to change the icon. The way to prevent
+            // the user from typing text in <i> tags should be improved.
             if (
-                (isMediaElement(node) || node.tagName === 'HR') &&
+                ((isMediaElement(node) && node.getAttribute('contenteditable') !== 'true') ||
+                node.tagName === 'HR') &&
                 node.getAttribute('contenteditable') !== 'false'
             ) {
                 node.setAttribute('contenteditable', 'false');
```

Comment:
@qsm-odoo I'm afraid my solution is imperfect here...
A good way to find out what is marked as an `editableArea` is to check if the `contenteditable` attribute is set to true. But then I feel like this whole condition is meaningless. Maybe we can stick to the solution I propose here for now and then improve this part in the future task that will fix this `contenteditable` mechanism what do you think ?

Reply:
I'll think about it, I am waiting for some editor team feedback. They were speaking about adding a class independent from contenteditable to mark media elements that should be editable. Maybe we'll use that to fix this problem here if it works.

Reply:
Was there any update about this? 
Apparently the bug is still there. Is that still about the contentediable issue of the web editor :grimacing: ?

Reply:
I recently ping them about that, I'll ask again :wink:

Reply:
I know this bug isn't a major one, but is there any news about it? I've had two tasks blocked because of this for over 4 months (this one and the before and after snippet). If I can do anything to help tell me :)

Reply:
@Guillaume-gdi Could you revive this PR as soon as possible? :+1: The editor team fix (which I reviewed a lot in the end) was merged, here is the 16.2 version: https://github.com/odoo/odoo/commit/6ead397f33512c918f51feb2b32fb3e15ffedf34 -> you now have a class which you can use to mark editable media in non editable environments. See how the company team snippet was adapted (both in stable and master).

Note that I also have a follow-up for master with https://github.com/odoo/odoo/pull/131357 but this is proving annoying to merge for now :grimacing: 

If I am not mistaken, [this issue](https://github.com/odoo/odoo/issues/104048) is related to your PR, don't forget to mention it in the commit message with "Fixes XXX".

Reply:
@qsm-odoo , I see that the issue has been fixed in the meantime with [this commit](https://github.com/odoo/odoo/commit/bb291b35e46e42ca098aa3e04ac0b473f72be423). Do you want me to revert the commit before doing the fix with the new class ?

Reply:
Strange... you started this conversation the end of the december and we talked about it for months afterwards... how did we not see it was fixed early december? x)

Anyways, sure, their fix is not good. First, the TODO comment that was there before still make sense to me: we should not bypass what the double click does. If anything, we should refactor the system so that if the double click does not work, then clearly the "Replace" button should not be shown. You can just restore that TODO comment for now. Second, I guess their fix only fixed the "Replace" button... but the double click should work, so yes, we should use the new system.

(Don't make a dedicated revert commit, do everything in one commit :+1:)